### PR TITLE
Fix typos in docs and comments

### PR DESCRIPTION
**Pull request by szkarpinski**
## Category:

**Refactoring** (*Redesign of existing code that doesn't affect functionality*)

## Description:
In this PR I fix some typos in DALI :)

This work was done in **11 minutes** using a tool from #5193 

### Tests:

- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [x] Docstring
  - [x] Doxygen
  - [x] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/python/nvidia/dali/plugin/base_iterator.py [hunk 1]
```diff
@@ -278,7 +278,7 @@ def check_all_or_none_and_get(input_meta, name, err_msg):
             self._shard_sizes_per_gpu = self._calculate_shard_sizes(np.arange(0, self._shards_num))
 
             # to avoid recalculation of shard sizes when iterator moves across the shards
-            # memorize the initial shard sizes and then use chaning self._shards_id to index it
+            # memorize the initial shard sizes and then use changing self._shards_id to index it
```
Comment (JanuszL) [inline dali/python/nvidia/dali/plugin/base_iterator.py:281]:
```suggestion
            # memorize the initial shard sizes and then change self._shards_id to index it
``?

File: docs/examples/use_cases/pytorch/efficientnet/readme.rst [hunk 2]
```diff
@@ -133,19 +133,19 @@ To run training benchmarks with different data loaders and automatic augmentatio
   export RESULT_WORKSPACE=./
 
   # synthetic benchmark
-  python multiproc.py --nproc_per_node 8 ./main.py --amp --static-loss-scale 128 --batch-size 128 --epochs 1 --prof 1000 --no-checkpoints --training-only --data-backend synthetic --workspace $RESULT_WORKSPACE --raport-file bench_report_synthetic.json $PATH_TO_IMAGENET
+  python multiproc.py --nproc_per_node 8 ./main.py --amp --static-loss-scale 128 --batch-size 128 --epochs 1 --prof 1000 --no-checkpoints --training-only --data-backend synthetic --workspace $RESULT_WORKSPACE --report-file bench_report_synthetic.json $PATH_TO_IMAGENET
```
Comment (JanuszL) [inline docs/examples/use_cases/pytorch/efficientnet/readme.rst:136]: `docs/examples/use_cases/pytorch/efficientnet/main.py` `"--raport-file"` needs to be adjusted as well.

File: dali/python/nvidia/dali/plugin/base_iterator.py (hunk 1 above)
Comment (szkarpinski) [reply dali/python/nvidia/dali/plugin/base_iterator.py:281]: I looked into the code and the logic there is that `size._shards_id` is used as an index of `self._shard_sizes_per_gpu` and `size._shards_id` is changing frequently. So the meaning here was to _use self._shards_id, which will be changing frequently_, so `size._shards_id` is _used_ and is _changing_.

File: docs/examples/use_cases/pytorch/efficientnet/readme.rst (hunk 2 above)
Comment (szkarpinski) [reply docs/examples/use_cases/pytorch/efficientnet/readme.rst:136]: Oh, nice catch, thank you! Fixed in https://github.com/NVIDIA/DALI/pull/5194/commits/1abfa38d0c678771629ed8646a7e735824a683f1 . 

Btw @klecki, just to confirm, are we OK with breaking "backward compatibility" of this example?

Comment (JanuszL) [reply docs/examples/use_cases/pytorch/efficientnet/readme.rst:136]: @stiepan ?

**Review by JanuszL — APPROVED** [review]

File: dali/operators/reader/coco_reader_op.cc [hunk 3]
```diff
@@ -69,7 +69,7 @@ This readers produces the following outputs::
      ...
      [mask_idx_P, start_vertex_idx_P, end_vertex_idx_P]]
 
-  where ``mask_idx`` is the index of the mask the polygon, in the range ``[0, M)``, and ``start_vertex_idx`` and  ``end_verted_idx``
+  where ``mask_idx`` is the index of the mask the polygon, in the range ``[0, M)``, and ``start_vertex_idx`` and  ``end_vertex_idx``
```
Comment (awolant) [inline dali/operators/reader/coco_reader_op.cc:72]: Not sure about this one, looks like it should stay as it was

**Review by awolant — APPROVED** [review]

File: docs/examples/use_cases/pytorch/efficientnet/readme.rst (hunk 2 above)
Comment (stiepan) [reply docs/examples/use_cases/pytorch/efficientnet/readme.rst:136]: I am not aware of some external automation that would rely on it. I think we should be good with the fix.

Comment (szkarpinski) [reply docs/examples/use_cases/pytorch/efficientnet/readme.rst:136]: Thank you!

**Comment by szkarpinski** [comment]
!build

**Comment by szkarpinski** [comment]
!build
